share/spice/currency/currency.js

```diff
@@ -287,13 +287,15 @@
             }
 
             var endpoint = "/js/spice/currency/1/" + from + "/" + to;
-            $.get(endpoint, function(payload) {
-
-                // jsonp is returned from the API so we have to alter the contents
-                var response = JSON.parse(payload.trim().replace(/^[^\(]*\(/, '').replace(/\);$/, ''));
-                response = response.conversion;
-                Converter.resetConverter(response);
-
+            $.ajax({ 
+                url: endpoint,
+                dataType: 'text',
+                success: function(payload) {
+                    // jsonp is returned from the API so we have to alter the contents
+                    var response = JSON.parse(payload.trim().replace(/^[^\(]*\(/, '').replace(/\);$/, ''));
```

Comment:
Should we maybe make a way to avoid this in `DDG::Rewrite` -- namely a way to return only JSON? I can see ignoring that though. In any case, it seems like we'd need to update some of the above.

Reply:
That would be the most appropriate solution, that would mean `dataType: 'text'` could then be `dataType: 'json'` and this replace logic wouldn't be needed. Not sure how to go about makign the Perl changes, though...